File: woodroid/src/main/java/me/gilo/woodroid/data/api/ReportAPI.java
package me.gilo.woodroid.data.api;


import me.gilo.woodroid.models.report.*;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

import java.util.List;
import java.util.Map;

public interface ReportAPI {

    @GET("reports/sales")
    Call<List<SalesTotal>> sales();

    @GET("reports/sales")
    Call<List<SalesTotal>> sales(@QueryMap Map<String, String> filter);


    @GET("reports/top_sellers")
    Call<List<TopSellerProducts>> top_sellers();

    @GET(" /wp-json/wc/v3/reports/top_sellers")
    Call<List<TopSellerProducts>> top_sellers(@QueryMap Map<String, String> filter);


    @GET("reports/coupons/totals")
    Call<List<CouponsTotal>> coupons_totals();


    @GET("reports/customers/totals")
    Call<List<CustomersTotal>> customers_totals();


    @GET("reports/orders/totals")
    Call<List<OrdersTotal>> orders_totals();

    @GET("reports/products/totals")
    Call<List<ProductsTotal>> products_totals();

    @GET("reports/reviews/totals")
    Call<List<ReviewsTotal>> reviews_totals();

}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/ReportAPI.kt
package me.gilo.woodroid.data.api


import me.gilo.woodroid.models.report.*
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.QueryMap

interface ReportAPI {

    @GET("reports/sales")
    fun sales(): Call<List<SalesTotal>>

    @GET("reports/sales")
    fun sales(@QueryMap filter: Map<String, String>): Call<List<SalesTotal>>


    @GET("reports/top_sellers")
    fun top_sellers(): Call<List<TopSellerProducts>>

    @GET(" /wp-json/wc/v3/reports/top_sellers")
    fun top_sellers(@QueryMap filter: Map<String, String>): Call<List<TopSellerProducts>>


    @GET("reports/coupons/totals")
    fun coupons_totals(): Call<List<CouponsTotal>>


    @GET("reports/customers/totals")
    fun customers_totals(): Call<List<CustomersTotal>>


    @GET("reports/orders/totals")
    fun orders_totals(): Call<List<OrdersTotal>>

    @GET("reports/products/totals")
    fun products_totals(): Call<List<ProductsTotal>>

    @GET("reports/reviews/totals")
    fun reviews_totals(): Call<List<ReviewsTotal>>

}